src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1505,19 +1512,32 @@ def status(self, id):
 
         return vnc_web
 
-    def __del__(self):
+    @private
+    def close_libvirt_connection(self):
         if self.libvirt_connection:
             with contextlib.suppress(libvirt.libvirtError):
                 self.libvirt_connection.close()
+            self.libvirt_connection = None
+
+    @private
+    async def terminate(self):
+        await SHUTDOWN_LOCK.acquire()
```

Comment:
At this point probably doesnt matter much, but for completeness should prob use:
```
async with SHUTDOWN_LOCK:
```
?